packages/kn-plugin-workflow/pkg/common/containers.go

```diff
@@ -205,7 +205,7 @@ func pullDockerImage(cli *client.Client, ctx context.Context) (io.ReadCloser, er
 	// of an image.
 	imageNameWithoutRegistry := strings.Split(metadata.DevModeImage, "/")
 	imageFilters := filters.NewArgs()
-	imageFilters.Add("reference", fmt.Sprintf("*/%s", imageNameWithoutRegistry[len(imageNameWithoutRegistry)-1]))
+	imageFilters.Add("reference", fmt.Sprintf("*/*/%s", imageNameWithoutRegistry[len(imageNameWithoutRegistry)-1]))
```

Comment:
This should be fixed upstream, too, and the client must comply with images that are in the format <namespace>/<name>:<tag> and also <remote>/<namespace>/<name>:<tag>. Does this fix consider this scenario? Please open an issue upstream and fix it there, too, so we will have a clean branch for 1.35.